jwst/outlier_detection/outlier_detection.py

```diff
@@ -406,6 +407,13 @@ def detect_outliers(self, blot_models):
             for i in range(len(self.input_models)):
                 self.inputs.dq[i, :, :] = self.input_models[i].dq
 
+            # Make sure all data arrays have matching NaNs and DQ flags
+            match_nans_and_flags(self.inputs)
```

Comment:
To check that I'm following the changes. With this change calling outlier detection will now also assign `nan` to all pixels marked as outliers (whereas before it only updated the dq flags for those pixels)?

Reply:
Yes, that's correct.

Reply:
Thanks! I didn't quite get that from the ticket which mentions setting err and dq when sci is nan or sci and dq when err is nan (but not setting sci and err when dq is do not use).